packages/cht/test/Adaptor.test.js

```diff
@@ -0,0 +1,206 @@
+import { expect } from 'chai';
+import { enableMockClient } from '@openfn/language-common/util';
+
+import { request, get, post, put } from '../src/Adaptor.js';
+
+const testServer = enableMockClient('https://example.cht.com');
+
+describe('request', () => {
+  it('should send a GET request', async () => {
+    const data = {
+      has_permissions: true,
+    };
+
+    testServer
+      .intercept({
+        path: '/api/v1/settings',
+        method: 'GET',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+      .reply(200, data, { headers: { 'content-type': 'application/json' } });
+
+    const state = {
+      configuration: {
+        baseUrl: 'https://example.cht.com',
+        username: 'hello',
+        password: 'there',
+      },
+      data,
+    };
+
+    const finalState = await request('GET', '/api/v1/settings')(state);
+
+    expect(finalState.data).to.eql(data);
+  });
+
+  it('should throw for a 403 response', async () => {
+    testServer
+      .intercept({
+        path: '/api/v1/contacts',
+        method: 'GET',
+      })
+      .reply(403);
+
+    const state = {
+      configuration: {
+        baseUrl: 'https://example.cht.com',
+        username: 'hello',
+        password: 'there',
+      },
+    };
+
+    const error = await request(
+      'GET',
+      '/api/v1/contacts'
+    )(state).catch(error => {
+      return error;
+    });
+    expect(error.statusMessage).to.eql('Forbidden');
+  });
+});
+
+describe('get()', () => {
+  it('should make a simple get request', async () => {
+    const data = {
+      has_permissions: true,
+    };
+
+    testServer
+      .intercept({
+        path: '/api/v1/settings',
+        method: 'GET',
+        headers: {
+          Authorization: 'Basic aGVsbG86dGhlcmU=',
+        },
+      })
+      .reply(200, data, { headers: { 'content-type': 'application/json' } });
+
+    const state = {
+      configuration: {
+        baseUrl: 'https://example.cht.com',
+        username: 'hello',
+        password: 'there',
+      },
+      data,
+    };
+
+    const finalState = await get('/api/v1/settings')(state);
+
+    expect(finalState.data).to.eql(data);
+    expect(finalState.response.method).to.eql('GET');
+  });
+  it('should make a get with parameters', async () => {
```

Comment:
This test doesn't seem right...

1) there are no query parameters
2) it's returning XML for some reason

I'm quite happy to include an XML test, but put it under `request` tests and call it, like, `should return XML responses as a string"

The get test here needs to be like `get('/whatever', { query: { /*something*/ }  })`